packages/client/src/components/UserContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { gameIO } from 'lib/socket';
+import { useEffect } from 'react';
+import { useState } from 'react';
+import { FC, createContext, useContext } from 'react';
+import { getPersistedUserName, persistUserName } from 'utils/localStorageUtils';
+
+export const UserContext = createContext<{
+  userName: string | null;
+}>({
+  userName: null,
+});
+
+const persistedUserName = getPersistedUserName();
```

Comment:
You have user both `username` and `userName`. I prefer the former, username is usually one word. So that would have to change in a few places.